### ISSUE TITLE
unknown format \"X\" is used in schema at path

### ISSUE THREAD

**Issue by EdwardSalter** [jormaechea/open-api-mocker#19]
I am receiving the following error when trying to make a call to the mock API using a request body that contains any number format:
```yaml
type: object
properties:
  branchId:
    type: number
    format: double # Any value here seems to cause an error
```

The error I am receiving is:
> "unknown format \"double\" is used in schema at path \"#/properties/branchId\""

I have attempted to tried various combinations of `type: number` and `type: integer` against each of the number formats defined [in the spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.3.md#data-types).
The spec also outlines that the `format` field is optional and can take any value.
It seems wrong that this error is then occurring due to a field that is considered optional.

In terms of fixing this, I see that there is a [line of code in the schema-valaidator](https://github.com/jormaechea/open-api-mocker/blob/bfce991538510af11b0db378945dd55197fc55f2/lib/schema-validator/index.js#L5) that adds the `password` format to the `unknownFormats` array used by _ajv_, is it a case of just adding in all the recognised formats to that array (`["int32", "int64", "float", "double", "byte", "binary", "date", "date-time", "password"]`)?
Whilst this would fix the problems for those schemas that use those default formats, any schema that uses (a completely valid) format that is not in that list will still throw this error. To prevent any further errors for formats, is it best to add some other kind of input to allow more unknown formats to be specified?

**Comment by EdwardSalter** [comment]
Looking into what `ajv` actually does since I am not familiar with it it looks like it supports many different options that handle the `format` property.
Out of the box it looks like it supports all formats out of the box and also includes the ability to add in new ones: https://ajv.js.org/guide/formats.html#string-formats

Perhaps we could add in some new validation functions for those basic formats as described by the OpenAPI spec to  AJV. 
As for custom formats, the only real way I can think of supporting that via using the binary file directly is to provide a new command line argument: `--unknownFormats=customFormat1,customFormat2` and these values simply get passed to the SchemaValidator. If calling the code directly, we could add some form of API (e.g. a new fluent method or something) to the OpenApiMocker class itself that allows custom validation functions to be specified.
What are your thoughts on this?

**Comment by jormaechea** [comment]
Released in v1.6.0 🚀
